### DISCUSSION TITLE
Question about mmseg pre-trained weight loading

### DISCUSSION THREAD

**Discussion by endaoguansanlu** [open-mmlab/mmcv#2354] [General]
Hello, can mmseg load two pre-training weights, for example, a two-branch model, one branch is segformer, one branch is resnet, I will load the two pre-training weights to this two-branch model and then perform it on cityscapes segmentation.Looking forward to your reply, thank you!

**Comment by HAOCHENYE** [comment]
Hi, you can load different pre-trainied weights as the tutial shows:
https://mmengine.readthedocs.io/zh_CN/latest/advanced_tutorials/initialize.html#id3

We highly recommend you to experience `MMEngine`, which is next-generation training framework of Openmmlab. Since we haven't made many changes to the initialization module yet, the content of the tutorial also applies to MMCV.

Since `SegFormer` and `ResNet` are also the subclass of `BaseModules`, you can define your model like this:

```python
model=dict(
    branch1=dict(type='SegFormer', init_cfg=dict(type='Pretrained', checkpoint='path/to/checkpoint1'))
    branch2=dict(type='ResNet', init_cfg=dict(type='Pretrained', checkpoint='path/to/checkpoint2'))
) 
```
